ui/src/pages/Create.js

```diff
@@ -0,0 +1,645 @@
+        ? <span>{this.state.formPhotoFile.name}</span>
+        : null;
+    var photoDeleteButton = null;
+    if (this.state.formPhotoFile) {
+      photoDeleteButton = (
+        <span onClick={(e) => {this.setState({formPhotoFile: ''})}}>X</span>
+      );
+    }
+    return (
+      <PFNotchedOutline label={this.props.intl.formatMessage(messages.photo)}>
+        <input
+          type='file'
+          ref={this.photoUploadInput}
+          onChange={(e) => this.setState({formPhotoFile: e.target.files[0]})} />
+        <Button
+          className='pf-button-primary'
+          type='button'
+          onClick={() => this.photoUploadInput.current.click()}
+          disabled={this.state.formPhotoUrl != ""}>
+          {this.props.intl.formatMessage(messages.uploadPhoto)}
+        </Button>
+        {photoFilenameSpan}
+        {photoDeleteButton}
+        <TextField
+          label={this.props.intl.formatMessage(messages.orEnterPhotoUrl)}
+          outlined
+        >
+          <Input
+            value={this.state.formPhotoUrl}
+            onChange={(e) => this.setState({formPhotoUrl: e.target.value})}
+            disabled={this.state.formPhotoFile != ""} />
+        </TextField>
+      </PFNotchedOutline>
+    );
+  }
+
+  /*
+   * Adds a profile page field for the given site to the form.
+   */
+  addProfilePageField(site) {
+    this.setState({
+      formProfilePages: this.state.formProfilePages.concat(new Map(
+          [['site', site], ['value', '']])),
+      showProfilePageOptions: false,
+    });
+  }
+
+  /*
+   * Removes the given profile page field from the form.
+   */
+  removeProfilePageField(index) {
+    var newProfileList = this.state.formProfilePages.slice(0, index);
+    if (index < this.state.formProfilePages.length - 1) {
+      newProfileList = newProfileList.concat(
+          this.state.formProfilePages.slice(
+              index + 1, this.state.formProfilePages.length));
+    }
+    this.setState({
+      formProfilePages: newProfileList,
+    });
+  }
+
+  /*
+   * Renders a profile field menu option for the given site.
+   */
+  renderProfilePageOption(site) {
+    // TODO(nworden): include site icons
+    return (
+      <li
+        className='mdc-typography--body1'
+        onClick={() => this.addProfilePageField(site)}>
+        <FormattedMessage {...CREATE_PROFILE_PAGE_SITES[site]} />
+      </li>
+    );
+  }
+
+  /*
+   * Render the profile page link fields (if any) and the "Add site" button.
+   */
+  renderProfilePageFields() {
+    var profilePageFields = this.state.formProfilePages.map((page, index) => (
+      <div
+        className='create-profilefieldwrap'
+        key={this.profileFieldIdCounter++}>
+        <TextField
+          label={this.props.intl.formatMessage(
+              CREATE_PROFILE_PAGE_SITES[page.get('site')])}
+          outlined
+        >
+          <Input
+            value={this.state.formProfilePages[index].value}
+            onChange={(e) => {
```

Comment:
Would be better to extract this function as a method instead of embedding it inside HTML.